webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -418,35 +469,41 @@ class ClassicTechniqueWriter(basePath : String, parameterTypeService: ParameterT
 
     val bundleParams = if (technique.parameters.nonEmpty) technique.parameters.map(_.name.canonify).mkString("(",",",")") else ""
 
-    val methodCalls =
-      ( for {
-        (method,index) <- technique.methodCalls.zipWithIndex
-        method_info <- methods.get(method.methodId)
-        (_, classParameterValue) <- method.parameters.find( _._1 == method_info.classParameter)
-
-        params <- Control.sequence(method_info.parameters) {
-          p =>
-            import  com.normation.box.EitherToBox
-            for {
-              (_,value) <- Box(method.parameters.find(_._1 == p.id))
-              escaped   <- parameterTypeService.translate(value, p.parameterType, AgentType.CfeCommunity).toBox
-            } yield {
-              escaped
+    def bundleMethodCall( parentBlocks : List[MethodBlock])(method : MethodElem) : List[String] = {
+      method match {
+        case call : MethodCall =>
+          (for {
+            method_info <- methods.get(call.methodId)
+            (_, classParameterValue) <- call.parameters.find( _._1 == method_info.classParameter)
+
+            params <- Control.sequence(method_info.parameters) {
+              p =>
+                import  com.normation.box.EitherToBox
+                for {
+                  (_,value) <- Box(call.parameters.find(_._1 == p.id))
+                  escaped   <- parameterTypeService.translate(value, p.parameterType, AgentType.CfeCommunity).toBox
+                } yield {
+                  escaped
+                }
             }
-        }
-
-      } yield {
-        val condition = canonifyCondition(method)
-        val promiser = s"${escapeCFEngineString(method.component)}_$${report_data.directive_id}_${index}"
-        // Check constraint and missing value
-        val args = params.mkString(", ")
-
-        s"""    "${promiser}" usebundle => ${reportingContext(method, classParameterValue)},
-           |     ${promiser.map(_ => ' ')}         if => concat("${condition}");
-           |    "${promiser}" usebundle => ${method.methodId.value}(${args}),
-           |     ${promiser.map(_ => ' ')}         if => concat("${condition}");""".stripMargin('|')
-
-      }).mkString("\n")
+          }  yield {
+            val condition = canonifyCondition(call, parentBlocks)
+            val promiser = call.id
```

Comment:
I don't understand where the ID is generated.

Reply:
ok, it's in `webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/MethodElemUtils.elm`